cookbook/core/flyte_basics/files.py

```diff
@@ -2,86 +2,138 @@
+Files are one of the most fundamental entities that users of Python work with,
+and they are fully supported by Flyte. In the IDL, they are known as
+`Blob <https://github.com/flyteorg/flyteidl/blob/cee566b2e6e109120f1bb34c980b1cfaf006a473/protos/flyteidl/core/literals.proto#L33>`__
+literals which are backed by the
+`blob type <https://github.com/flyteorg/flyteidl/blob/cee566b2e6e109120f1bb34c980b1cfaf006a473/protos/flyteidl/core/types.proto#L47>`__.
+
+Let's assume our mission here is pretty simple. We download a few csv file
+links, read them with the python built-in :py:func:`csv.DictReader` function,
+normalize some pre-specified columns, and output the normalized columns to
+another csv file.
 
-Let's assume our mission here is pretty simple. We take in a couple of links, download the pictures, rotate them, and return the rotated images.
 """
 
 # %%
 # First, let's import the libraries.
+import csv
 import os
+from collections import defaultdict
+from typing import List
 
-import cv2
 import flytekit
 from flytekit import task, workflow
-from flytekit.types.file import JPEGImageFile
+from flytekit.types.file import FlyteFile
+
 
 # %%
-# ``JPEGImageFile`` is a pre-formatted FlyteFile type. It is equivalent to ``FlyteFile[typing.TypeVar("jpeg")]``.
+# Next, we write a task that accepts a ``FlyteFile``, a list of column names,
+# and a list of column names to normalize, then outputs a csv file of only
+# the normalized columns. For this example we'll use z-score normalization,
+# i.e. mean-centering and standard-deviation-scaling.
 #
 # .. note::
-#   The ``FlyteFile`` literal can be scoped with a string, which gets inserted into the format of the Blob type ("jpeg" is the string in
-#   ``FlyteFile[typing.TypeVar("jpeg")]``). The format is entirely optional, and if not specified, defaults to ``""``.
-
+#   The ``FlyteFile`` literal can be scoped with a string, which gets inserted
+#   into the format of the Blob type ("jpeg" is the string in
+#   ``FlyteFile[typing.TypeVar("jpeg")]``). The format is entirely optional,
+#   and if not specified, defaults to ``""``.
 
-# %%
-# Next, we write a task that accepts ``JPEGImageFile`` as an input and returns the rotated image as an output,
-# which again is the ``JPEGImageFile``.
-# Files do not have a native object in Python, so we had to write one ourselves.
-# There does exist the ``os.PathLike`` protocol, but nothing implements it.
 @task
-def rotate(image_location: JPEGImageFile, location: str) -> JPEGImageFile:
-    """
-    Download the given image, rotate it by 180 degrees
-    """
-    working_dir = flytekit.current_context().working_directory
-    image_location.download()
-    img = cv2.imread(image_location.path, 0)
-    if img is None:
-        raise Exception("Failed to read image")
-    (h, w) = img.shape[:2]
-    center = (w / 2, h / 2)
-    mat = cv2.getRotationMatrix2D(center, 180, 1)
-    res = cv2.warpAffine(img, mat, (w, h))
+def normalize_columns(
+    csv_url: FlyteFile,
```

Comment:
Why a separate function? Keeping it the same makes it look as changes are minimal

Reply:
Not sure I understand the feedback 🤔... what separate function are you referring to?

Reply:
normalize_columns

Reply:
Still confused. Renamed the task from "rotate" to "normalize columns" because the use case has changed. It's still a task